server/core/src/main/java/dev/slimevr/protocol/ProtocolAPI.java

```diff
@@ -31,6 +32,10 @@ public ProtocolAPI(VRServer server) {
 	}
 
 	public void onMessage(GenericConnection conn, ByteBuffer message) {
+		if (message.position() != 0)
+			message = ByteBuffer
+				.wrap(Arrays.copyOfRange(message.array(), message.position(), message.limit()));
```

Comment:
Can we not use `ByteBuffer.slice()` here? If required, we can also further do `ByteBuffer.asReadOnlyBuffer()` but I believe it should be fine without.
```suggestion
		if (message.position() != 0)
			message = message.slice();
```

Reply:
I should add a comment explaining why this is necessary. The issue isn't mutability, it's Flatbuffers generated code failing to account for starting position properly. By using `Arrays.copyOfRange()` I allocate a new backing array without the data offset.

Reply:
Slice should set the starting position to 0 though, so it should work the same? I don't think flatbuffers ever directly accesses the backing array.

Reply:
Hmm... I know enough to know I don't know enough. What I do know is that packets fail to parse without the `copyOfRange()` ー blocking Monado from starting entirely ー and that `slice()`, `duplicate()`, and `asReadOnlyBuffer()` don't help.

Reply:
Okay but this needs to be investigated, otherwise it will create so much garbage 😢

Reply:
Turns out this was just a skill issue on my part. Changing `dst.slice()` to `dst.duplicate()` in `UnixSocketConnection.read()` fixed it.